scenes/game_elements/characters/enemies/guard/components/guard.gd

```diff
@@ -504,22 +504,28 @@ func _set_alert_other_sound_stream(new_value: AudioStream) -> void:
 
 
 func _on_instant_detection_area_body_entered(body: Node2D) -> void:
+	if not body is Player:
+		return
 	state = State.ALERTED
-	player_detected.emit(body)
+	player_detected.emit(body as Player)
```

Comment:
I believe that an `is` check narrows the type of the variable; i.e. I believe that inside an `if` block that has checked the type of `body` to be `Player`, then `body` has the type `Player`. Can you try this and see if it type-checks?

```gdscript
if body is Player:
    state = State.ALERTED
    player_detected.emit(body)
```

Reply:
The funny thing is, I can't get any error or warning to be logged by Godot for passing the wrong type of argument to `.emit()`!

Reply:
Yes, it wasn't warning before as well.